include/nlohmann/json.hpp

```diff
@@ -5371,8 +5371,18 @@ class basic_json // NOLINT(cppcoreguidelines-special-member-functions,hicpp-spec
         }
 
         // add element to array (move semantics)
+        const auto capacity = m_value.array->capacity();
         m_value.array->push_back(std::move(val));
-        set_parent(m_value.array->back());
+        if (capacity == m_value.array->capacity())
+        {
+            // capacity has not changed: updating parent of last element is sufficient
+            set_parent(m_value.array->back());
```

Comment:
Is this code in enough places that it makes sense to have a helper here?
One thought I had was adding an `old_capacity` parameter with a `-1` default to `set_parent` and have it fall back to `set_parents()` if the capacity was specified and changed.  That would allow you to continue using the return value of `set_parent`.
Not sure if that is cleaner or not.

Reply:
Yes, that code occurs in 3 places. I'll check.

Reply:
I refactored it. Feels clearer now.